dashboard/app/models/user.rb

```diff
@@ -615,7 +615,7 @@ def email_required?
   end
 
   def username_required?
-    provider == User::PROVIDER_MANUAL
+    provider == User::PROVIDER_MANUAL || username_changed?
```

Comment:
It seems strange to me to gate the username being required on whether the username is changed. Perhaps L366 should be `... if: :username_required? && username_changed?`?

Is there a reason we shouldn't require a username for all users? Note that there are roughly 3K production accounts with NULL username, most with NULL provider and some with google_oauth2 provider.